src/Core/Grid/Position/PositionDataHandler.php

```diff
@@ -0,0 +1,151 @@
+<?php
+/**
+ * 2007-2018 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Grid\Position;
+
+use PrestaShop\PrestaShop\Core\Grid\Position\Exception\PositionDataException;
+
+/**
+ * Class PositionDataHandler is a basic implementation of the PositionDataHandlerInterface,
+ * it transforms the provided array data into a PositionUpdate object.
+ * Of course, you need to respect a data format for it to work but if you don't want to use
+ * this format you are free to implement your own interface.
+ */
+class PositionDataHandler implements PositionDataHandlerInterface
```

Comment:
not a huge fan of names like "data handler/manager/...er" see my next comment for my proposal :)